::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
: Copyright (c) 2017 Enea AB and others.
:
: All rights reserved. This program and the accompanying materials
: are made available under the terms of the Apache License, Version 2.0
: which accompanies this distribution, and is available at
: http://www.apache.org/licenses/LICENSE-2.0
::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Thu, 19 Jan 2017 19:19:28 +0100
Subject: [PATCH] AArch64: nova: libvirt: Use host-model cpu

Closes-bug: https://jira.opnfv.org/browse/ARMBAND-193

Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
---
 deployment/puppet/openstack_tasks/manifests/roles/compute.pp | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/deployment/puppet/openstack_tasks/manifests/roles/compute.pp b/deployment/puppet/openstack_tasks/manifests/roles/compute.pp
index a6f4729..f6e6698 100644
--- a/deployment/puppet/openstack_tasks/manifests/roles/compute.pp
+++ b/deployment/puppet/openstack_tasks/manifests/roles/compute.pp
@@ -319,9 +319,18 @@ class openstack_tasks::roles::compute {
   # guest OS by using a combination of CPU features and other parameters (such
   # as CPUID level) that don't work. Until these issues are fixed, it's a good
   # idea to avoid using host-model
+  # NOTE(armband): AArch64: Use host-model cpu
   # http://libvirt.org/formatdomain.html#elementsCPU
   # https://bugs.launchpad.net/mos/+bug/1618473
-  $libvirt_cpu_mode = 'none'
+  # https://jira.opnfv.org/browse/ARMBAND-193
+  if str2bool($::is_virtual) {
+    $libvirt_cpu_mode = 'none'
+  } else {
+    $libvirt_cpu_mode = $::architecture ? {
+      /(arm64|aarch64)/ => 'host-model',
+      default           => 'none',
+    }
+  }

   # Install / configure nova-compute

